types/tests/bootstrap.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    fs,
+    path::{Path, PathBuf},
+    process::Command,
+};
+
+type Result<T> = ::std::result::Result<T, Box<dyn std::error::Error + Send + Sync>>;
+
+#[test]
+fn bootstrap() {
+    let proto_files = &["proto/narwhal.proto"];
+    let dirs = &["proto"];
+
+    println!(std::env!("CARGO_MANIFEST_DIR"));
+
+    let out_dir = PathBuf::from(std::env!("CARGO_MANIFEST_DIR"))
+        .join("src")
+        .join("generated");
+
+    let config = prost_build::Config::default();
```

Comment:
We should set the `bytes` config like I do [here](https://github.com/MystenLabs/narwhal/pull/171/files#diff-21641a0d45596ea45e12276ade1b91b43e87d4883fd244e1d8769c818040acd2R20) that way we can avoid doing some extra allocations

Reply:
I am more than happy to switch back, but first two questions for you

- What allocations are you talking about here? 
- I tried switching but I was having issues converting to and from `Bytes` to `[u8; 32]`. Do you know a clever way to do this?